tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline/taxi_pipeline_regression_e2e_test.py

```diff
@@ -0,0 +1,142 @@
+# Lint as: python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""E2E Tests for taxi pipeline beam with stub executors."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import filecmp
+
+from typing import Text
+import tensorflow as tf
+
+from tfx.examples.chicago_taxi_pipeline import taxi_pipeline_beam
+from tfx.experimental.pipeline_testing import pipeline_recorder_utils
+from tfx.experimental.pipeline_testing import stub_component_launcher
+from tfx.orchestration import metadata
+from tfx.orchestration.beam.beam_dag_runner import BeamDagRunner
+from tfx.orchestration.config import pipeline_config
+
+class TaxiPipelineRegressionEndToEndTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(TaxiPipelineRegressionEndToEndTest, self).setUp()
+    self._test_dir = os.path.join(
+        os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR', self.get_temp_dir()),
+        self._testMethodName)
+    self._pipeline_name = 'beam_stub_test'
+    # This example assumes that the taxi data and taxi utility function are stored in
+    # tfx/examples/chicago_taxi_pipeline. Feel free to customize this as needed.
+    taxi_root = os.path.dirname(taxi_pipeline_beam.__file__)
+    self._data_root = os.path.join(taxi_root, 'data', 'simple')
+    self._module_file = os.path.join(taxi_root, 'taxi_utils.py')
+    self._serving_model_dir = os.path.join(self._test_dir, 'serving_model')
+    self._pipeline_root = os.path.join(self._test_dir, 'tfx', 'pipelines',
+                                       self._pipeline_name)
+    # Metadata path for recording successful pipeline run.
+    self._metadata_db_uri = os.path.join(self._test_dir, 'tfx', 'record',
+                                         'metadata.db')
+    # Metadata path for stub pipeline runs.
+    self._metadata_path = os.path.join(self._test_dir, 'tfx', 'metadata',
+                                       self._pipeline_name, 'metadata.db')
+    self._output_dir = os.path.join(self._test_dir, 'testdata')
```

Comment:
How about renaming `_output_dir` to `_recorded_output` or `_recorded_output_dir` to make it clearer?